Engine/RealTime/BacktestingRealTimeHandler.cs

```diff
@@ -111,7 +115,16 @@ public void SetTime(DateTime time)
             // poke each event to see if it has fired, be sure to invoke these in time order
             foreach (var scheduledEvent in GetScheduledEventsSortedByTime())
             {
-                scheduledEvent.Scan(time);
+                if (scheduledEvent.NextEventUtcTime > time)
```

Comment:
This line is saying if the event's next time is in the future, then we don't need to call `Scan` on it. Further, we can stop checking the rest of the list since they're all sorted.

The line in `ScheduledEvent.Scan(DateTime)` is saying, if this event is at the present or in the past, then it should fire. Interestingly,  the two lines are _exactly_ opposite in their conditioning, such that, if a time passes one condition, it is guaranteed to _never_ pass the other condition.